node/src/controllers/booksController.ts

```diff
@@ -0,0 +1,76 @@
+import 'dotenv/config';
+import { Request, Response } from 'express';
+import { pool } from '../middlewares/db';
+
+export interface BookInfo {
+  book_id : number;
+  title : string;
+  author : string;
+  is_available : boolean;
+}
+
+export const getBook = async (req: Request, res: Response): Promise<void> => {
+  const { available, author } = req.query as { available?: string; author?: string };
+
+  let availableParsed: boolean | undefined;
+  if (available !== undefined) {
+    const lower = available.toLowerCase();
+    if (lower === 'true' || lower === 'false') {
+      availableParsed = lower === 'true';
+    } else {
+      res.status(400).json({ error: "Invalid 'available' query parameter. Expected 'true' or 'false'." });
+      return;
+    }
+  }
+
+  const authorParsed: string | undefined = author;
+
+  let query = 'SELECT * FROM books';
+  const params: any[] = [];
+
+  if (availableParsed !== undefined) {
+    query += params.length > 0 ? ' AND' : ' WHERE';
+    query += ' is_available = $' + (params.length + 1);
+    params.push(availableParsed);
+  }
+
+  if (authorParsed) {
+    query += params.length > 0 ? ' AND' : ' WHERE';
+    query += " author ILIKE '%' || $" + (params.length + 1) + " || '%'";
+    params.push(authorParsed);
+  }
+
+  console.debug('Executing query:', query, params);
+
+  try {
+    const result = await pool.query(query, params);
+    const books: BookInfo[] = result.rows;
+    res.json(books);
+  } catch (error) {
+    console.error('Error fetching books:', error);
+    res.status(500).json({ error: 'Internal server error' });
```

Comment:
```suggestion
    res.status(500).json({ message: '적절한 문구', errors: []});
```
아니면 error 타입 자체를 src/types에 선언하고 갈 것